File: app/src/main/java/com/mdgd/pokemon/ui/error/MessageDialog.java
package com.mdgd.pokemon.ui.error;

import com.mdgd.mvi.fragments.FragmentContract;
import com.mdgd.mvi.fragments.HostedDialogFragment;

public abstract class MessageDialog<
        VIEW extends FragmentContract.View,
        VIEW_MODEL extends FragmentContract.ViewModel<VIEW>,
        HOST extends FragmentContract.Host
        > extends HostedDialogFragment<VIEW, VIEW_MODEL, HOST> {

    protected static final String KEY_TITLE = "key_title";
    protected static final String KEY_MSG = "key_msg";
    protected static final String KEY_TITLE_STR = "key_title_str";
    protected static final String KEY_MSG_STR = "key_msg_str";
    protected static final String KEY_TYPE = "key_type";
    protected static final int TYPE_INT = 1;
    protected static final int TYPE_STR = 2;

}

File: app/src/main/java/com/mdgd/pokemon/ui/error/MessageDialog.kt
package com.mdgd.pokemon.ui.error

import com.mdgd.mvi.fragments.FragmentContract
import com.mdgd.mvi.fragments.HostedDialogFragment

abstract class MessageDialog<
        VIEW : FragmentContract.View,
        VIEW_MODEL : FragmentContract.ViewModel<VIEW>,
        HOST : FragmentContract.Host>
    : HostedDialogFragment<VIEW, VIEW_MODEL, HOST>() {

    companion object {
        const val KEY_TITLE = "key_title"
        const val KEY_MSG = "key_msg"
        const val KEY_TITLE_STR = "key_title_str"
        const val KEY_MSG_STR = "key_msg_str"
        const val KEY_TYPE = "key_type"
        const val TYPE_INT = 1
        const val TYPE_STR = 2
    }
}
